staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -253,296 +247,41 @@ func TestListContinuationWithFilter(t *testing.T) {
-		Recursive:       true,
-	}
-	if err := store.GetList(ctx, "/", options, out); err != nil {
-		t.Fatalf("Unable to get second page: %v", err)
-	}
-	if len(out.Continue) == 0 {
-		t.Fatalf("No continuation token set")
-	}
-	validateResourceVersion := storagetesting.ResourceVersionNotOlderThan(lastRVString)
-	storagetesting.ExpectNoDiff(t, "incorrect second page", []example.Pod{*preset[1].storedObj}, out.Items)
-	if err := validateResourceVersion(out.ResourceVersion); err != nil {
-		t.Fatal(err)
-	}
-	continueFromThirdItem := out.Continue
-	resolvedResourceVersionFromThirdItem := out.ResourceVersion
-	out = &example.PodList{}
-	options = storage.ListOptions{
-		ResourceVersion: "0",
-		Predicate:       pred(1, continueFromThirdItem),
-		Recursive:       true,
-	}
-	if err := store.GetList(ctx, "/", options, out); err != nil {
-		t.Fatalf("Unable to get second page: %v", err)
-	}
-	if len(out.Continue) != 0 {
-		t.Fatalf("Unexpected continuation token set")
-	}
-	storagetesting.ExpectNoDiff(t, "incorrect third page", []example.Pod{*preset[2].storedObj}, out.Items)
-	if out.ResourceVersion != resolvedResourceVersionFromThirdItem {
-		t.Fatalf("Expected list resource version to be %s, got %s", resolvedResourceVersionFromThirdItem, out.ResourceVersion)
-	}
-}
-
-func newTestLeaseManagerConfig() LeaseManagerConfig {
-	cfg := NewDefaultLeaseManagerConfig()
-	// As 30s is the default timeout for testing in global configuration,
-	// we cannot wait longer than that in a single time: change it to 1s
-	// for testing purposes. See wait.ForeverTestTimeout
-	cfg.ReuseDurationSeconds = 1
-	return cfg
-}
-
-func newTestTransformer() value.Transformer {
-	return storagetesting.NewPrefixTransformer([]byte(defaultTestPrefix), false)
-}
-
-type clientRecorder struct {
-	reads uint64
-	clientv3.KV
-}
-
-func (r *clientRecorder) Get(ctx context.Context, key string, opts ...clientv3.OpOption) (*clientv3.GetResponse, error) {
-	atomic.AddUint64(&r.reads, 1)
-	return r.KV.Get(ctx, key, opts...)
-}
-
-func (r *clientRecorder) GetReadsAndReset() uint64 {
-	return atomic.SwapUint64(&r.reads, 0)
-}
-
-type setupOptions struct {
-	client        func(*testing.T) *clientv3.Client
-	codec         runtime.Codec
-	newFunc       func() runtime.Object
-	prefix        string
-	groupResource schema.GroupResource
-	transformer   value.Transformer
-	pagingEnabled bool
-	leaseConfig   LeaseManagerConfig
-
-	recorderEnabled bool
-}
-
-type setupOption func(*setupOptions)
-
-func withClient(client *clientv3.Client) setupOption {
-	return func(options *setupOptions) {
-		options.client = func(t *testing.T) *clientv3.Client {
-			return client
+			t.Fatal(err)
 		}
-	}
-}
-
-func withClientConfig(config *embed.Config) setupOption {
-	return func(options *setupOptions) {
-		options.client = func(t *testing.T) *clientv3.Client {
-			return testserver.RunEtcd(t, config)
+		if _, err := etcdClient.KV.Compact(ctx, int64(rv), clientv3.WithCompactPhysical()); err != nil {
```

Comment:
IIRC some integration tests somewhere end up needing compaction as well, so there might be utility in moving this whole thing to the testing package (?) for reuse.

Reply:
I think it makes sense - but for the sake of avoiding making individual changes too large, I would prefer making that in further steps of the refactoring (if that's ok for you).

Reply:
Yep, that sounds good - was just raising awareness.